min_heap_binary_tree.rb

```diff
@@ -15,6 +15,39 @@ def insert(node)
     set_min_heap_state
   end
 
+  def delete(node)
+    replacement_node = tree[-1]
+    delete_index = tree.find_index(node)
+
+    delete_node(delete_index)
+    parent_child_check(replacement_node, delete_index)
+  end
+
+  def find(target_node)
```

Comment:
I feel like we are still not the same page with the `find` approach.  

I thought you were going to go with the implementation where you:

- loop through the `tree` array until you find what you’re looking for
- then eject from the loop

Not a `Queue` implementation, since the `tree` already is the BFS queue result.  No?